instrumentor/controllers/instrumentationconfig/manager.go

```diff
@@ -35,39 +34,12 @@ func SetupWithManager(mgr ctrl.Manager) error {
 		return err
 	}
 
-	// Watch for Deployment changes using DeploymentReconciler
+	// Watch for Source changes to update InstrumentationConfig
 	if err := builder.
 		ControllerManagedBy(mgr).
-		Named("instrumentor-instrumentationconfig-deployment").
-		For(&appsv1.Deployment{}).
-		WithEventFilter(workloadReportedNameAnnotationChanged{}).
-		Complete(&DeploymentReconciler{
-			Client: mgr.GetClient(),
-			Scheme: mgr.GetScheme(),
-		}); err != nil {
-		return err
-	}
-
-	// Watch for StatefulSet changes using StatefulSetReconciler
-	if err := builder.
-		ControllerManagedBy(mgr).
-		Named("instrumentor-instrumentationconfig-statefulset").
-		For(&appsv1.StatefulSet{}).
-		WithEventFilter(workloadReportedNameAnnotationChanged{}).
-		Complete(&StatefulSetReconciler{
-			Client: mgr.GetClient(),
-			Scheme: mgr.GetScheme(),
-		}); err != nil {
-		return err
-	}
-
-	// Watch for DaemonSet changes using DaemonSetReconciler
-	if err := builder.
-		ControllerManagedBy(mgr).
-		Named("instrumentor-instrumentationconfig-daemonset").
-		For(&appsv1.DaemonSet{}).
-		WithEventFilter(workloadReportedNameAnnotationChanged{}).
-		Complete(&DaemonSetReconciler{
+		Named("instrumentor-instrumentationconfig-source").
+		For(&odigosv1alpha1.Source{}).
+		Complete(&SourceReconciler{
```

Comment:
@RonFed wdyt about doing this in the startlangdetection Source controller instead of adding a new controller here? It would be nice to converge toward a single controller for Sources where possible

I think startlangdetection will already pick up these changes, it just won't act on them because `requestOdigletToCalculateRuntimeDetails` only calls [k8sClient.Create](https://github.com/odigos-io/odigos/blob/5cac2ff25aab58a6b0ec0e186d9f7e88f2aba8c4/instrumentor/controllers/startlangdetection/workload_controllers.go#L70). But it could call `Update` to patch in the new service name.

Either way, the Source startlangdetection controller should also handle setting service name when a Source is created anyway

Reply:
Ah, I completely agree. Doing it in a separate refactor PR is smart, no need to overcomplicate this change